### ISSUE TITLE
Wrong zenith time

### ISSUE THREAD

**Issue by ssandon** [shred/commons-suncalc#13]
On some location, the noon time is off.

Normally it should just set in the middle between rise and set time (when rise and set happends) and should be the maximum altitude reached.

```
double latitude = 14.750438D;
double longitude = -61.173173D;

long timestamp = 1561969563530L;
Date date = new Date(timestamp);

System.out.println("date=" + date.toString());

Calendar calendar = Calendar.getInstance();
calendar.setTime(date);

System.out.println("calendar=" + getFormatedDate(calendar.getTime(), TimeZone.getDefault()) +  " " + TimeZone.getDefault().toString());

TimeZone timeZone = TimeZone.getTimeZone("America/Martinique");

calendar.setTimeZone(timeZone);

System.out.println("calendar=" + getFormatedDate(calendar.getTime(), timeZone) + " " + timeZone.toString());

SunTimes suntimes = SunTimes.compute().on(calendar).timezone(timeZone).midnight().at(latitude, longitude).execute();

System.out.println("suntimes=" + suntimes);

```
Output is :

date=Mon Jul 01 10:26:03 CEST 2019
calendar=lundi 1 juillet 2019 10:26 sun.util.calendar.ZoneInfo[id="Europe/Paris",offset=3600000,dstSavings=3600000,useDaylight=true,transitions=184,lastRule=java.util.SimpleTimeZone[id=Europe/Paris,offset=3600000,dstSavings=3600000,useDaylight=true,startYear=0,startMode=2,startMonth=2,startDay=-1,startDayOfWeek=1,startTime=3600000,startTimeMode=2,endMode=2,endMonth=9,endDay=-1,endDayOfWeek=1,endTime=3600000,endTimeMode=2]]
calendar=lundi 1 juillet 2019 04:26 sun.util.calendar.ZoneInfo[id="America/Martinique",offset=-14400000,dstSavings=0,useDaylight=false,transitions=5,lastRule=null]

suntimes=SunTimes[rise=Mon Jul 01 11:39:00 CEST 2019, set=Tue Jul 02 00:38:00 CEST 2019, noon=Mon Jul 01 17:35:00 CEST 2019, nadir=Mon Jul 01 06:05:00 CEST 2019, alwaysUp=false, alwaysDown=false]

computed noon is 11:35GTM-04:00 or 17:35GMT+02:00
it should in fact be around 12:08GTM-04:00 or 18:08GMT+02:00

could you check if I am right ?
Thank you.

**Comment by shred** [comment]
Yes, the noon (and I assume also the nadir) output is incorrect. I will have a look into it.

**Comment by shred** [comment]
The library uses quadratic interpolation for sunrise, sunset, noon, and nadir calculation. Under rare circumstances, the noon and nadir times were extrapolated though, giving a major error in the calculated time. I am now using 1 hour steps instead of 2 hour steps in SunTimes. It doubles the calculation time, but gives a better result.

Noon is at 12:06 now, which is a little off 12:08, but still in the tolerance that I would accept for this library.

Fixed in commit f71af9c. I will release a version 2.5 soon.

Thank you for reporting the issue! :smile:

**Comment by ssandon** [comment]
Thank you for your prompt answer.
I built the corrected code and tested it against my application. It gives
very acceptable results I think.

Best regards
Stéphane
